### PR TITLE
Adapt InstPreInstall to storage-ng

### PR DESCRIPTION
**Pull request by ancorgs**
This needs more manual testing (or even creating new automated tests) before being merged.

### CODE REVIEW

File: src/lib/installation/clients/inst_pre_install.rb [hunk 1]
```diff
@@ -225,13 +224,12 @@ def Initialize
...
         helper = lambda do |s|
           count = 0
-          disks = Builtins.filter(disks) do |dist|
-            next true if Builtins.search(dist, s) != 0
-            count = Ops.add(count, 1)
-            Ops.less_or_equal(count, 8)
+          disks = disks.select do |dist|
+            next true unless dist.start_with?(s)
+            count += 1
+            count <= 8
```
Comment (wfeldt) [inline src/lib/installation/clients/inst_pre_install.rb]: Not being a ruby expert this looks weird to me; what's wrong with simply `(count += 1) <= 8`?

File: src/lib/installation/clients/inst_pre_install.rb [hunk 2]
```diff
@@ -225,13 +224,12 @@ def Initialize
       # mounting of all partitions (fate#305873, bnc#468922)
       # FIXME: copy-pasted from partitioner, just different number of disks and added /dev/dasd
       restrict_disk_names = lambda do |disks|
```
Comment (wfeldt) [inline src/lib/installation/clients/inst_pre_install.rb]: After thinking about this for a while I'm still not happy with this concept. In particular as we don't seem to care at all for updates here:

https://github.com/yast/yast-update/blob/master/src/modules/RootPart.rb#L2066-L2069

If at all this functionality belongs to yast-storage-ng.

Issues I have with this concept:
- does not care about other device classes (vd*, dm*, and all the modern stuff)
- it makes only sense to cut the device list if the number of file systems exceeds a certain limit (say, - completely random - 32)

**Comment by wfeldt** [comment]
lgtm (but for the rubocop offense)

File: src/lib/installation/clients/inst_pre_install.rb (hunk 1 above)
Comment (ancorgs) [reply src/lib/installation/clients/inst_pre_install.rb]: Both are perfectly valid and totally equivalent and none of them look weird to my eyes. :) I'm fine with changing it if you dislike the original for any reason.

Comment (wfeldt) [reply src/lib/installation/clients/inst_pre_install.rb]: no need, I lgtm'ed below; was just some thought

**Pull request merged**